### PR TITLE
[BugFix] Preserve Delta multidimensional event shapes

### LINKED ISSUE #4090: Delta mishandles multidimensional event shapes
## Describe the bug

`torchrl.modules.distributions.Delta` mishandles an explicitly configured multidimensional `event_shape` in two adjacent paths:

1. `log_prob()` reduces changing negative dimensions and returns the wrong batch shape.
2. `expand()` silently reclassifies one event dimension as a batch dimension in the returned distribution.

## To reproduce

```python
import torch

from torchrl.modules.distributions import Delta


param = torch.arange(24.0).reshape(2, 3, 4)
dist = Delta(param, batch_shape=(2,), event_shape=(3, 4))
expanded = dist.expand((5, 2))

print("original:", dist.batch_shape, dist.event_shape, dist.log_prob(dist.param).shape)
print("expanded:", expanded.batch_shape, expanded.event_shape, expanded.param.shape)
print("expanded log_prob:", expanded.log_prob(expanded.param).shape)
```

On current `main` this prints:

```text
original: (2,) (3, 4) torch.Size([3])
expanded: torch.Size([5, 2, 3]) torch.Size([4]) torch.Size([5, 2, 3, 4])
expanded log_prob: torch.Size([5, 2, 3])
```

## Expected behavior

The original and expanded distributions should preserve the configured event semantics:

```text
original: (2,) (3, 4) torch.Size([2])
expanded: torch.Size([5, 2]) torch.Size([3, 4]) torch.Size([5, 2, 3, 4])
expanded log_prob: torch.Size([5, 2])
```

`Delta.expand()` currently creates the returned `Delta` without forwarding `batch_shape` or `event_shape`, so the constructor falls back to inferring only the final parameter dimension as the event. Separately, `_is_equal()` reduces `-1` and then `-2` after the tensor rank has changed; reducing the current last dimension for each event axis preserves the batch dimensions.

## System information

- TorchRL: current `main` at `7294465173cb23121386db0ce115abba54fc4a28`
- PyTorch: `2.11.0+cu130` (reproduced on CPU)
- TensorDict: `0.13.0`
- OS: Windows 11

## AI assistance disclosure

I used OpenAI Codex to help inspect the implementation, construct and run the pristine reproduction, and search existing issues and pull requests for duplicates. I reviewed and reproduced the behavior before submitting this report.

### PR DESCRIPTION
**Pull request by aswanth-07**
## Description

Preserve `Delta` distributions' multidimensional event semantics in both expansion and log-probability evaluation.

`Delta.expand()` now forwards the requested batch shape and existing event shape to the returned distribution instead of falling back to one-dimensional event inference. `_is_equal()` now reduces the current final dimension once per event axis, so `log_prob()` retains every batch dimension when the event has more than one axis.

An existing distribution test module now covers a `(2,)` batch with a `(3, 4)` event before and after expansion to `(5, 2)`. It also perturbs one element in a single batch item and verifies that the negative log-probability remains localized to that item before and after expansion.

## Motivation and Context

Without this change, `Delta(param, batch_shape=(2,), event_shape=(3, 4))` returns a `log_prob` with shape `(3,)`. Expanding it to batch shape `(5, 2)` also silently changes its metadata to batch shape `(5, 2, 3)` and event shape `(4,)`. The parameter and sample shapes remain plausible, which makes the semantic corruption easy to miss.

Closes #4090

- [x] I have raised an issue to propose this change: #4090

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Validation

- The focused regression was run against the pristine implementation and failed with the incorrect expanded batch shape.
- `python -m pytest test/test_distributions.py::TestDelta -q`: 11 passed
- Repository pre-commit hooks on both changed files: all passed
- `git diff --check`: passed

All local tests were run on CPU in a single process.

## Checklist

- [x] I have read the CONTRIBUTING guide and `CLAUDE.md`.
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly.
- [ ] I have updated the documentation accordingly.

## AI assistance disclosure

I used OpenAI Codex to help inspect the implementation, construct and run the pristine reproduction, search existing issues and pull requests for duplicates, implement the focused fix and regression, run validation, and draft this pull request. I reviewed the diff and reproduced the behavior before submission.

### CODE REVIEW

**Review by vmoens — APPROVED** [review]
LGTM thanks

**Comment by aswanth-07** [comment]
Thanks :)
